README.md

```diff
@@ -1,325 +1,293 @@
+
+You will also need [Node.js](https://nodejs.org/en/), [NPM](https://www.npmjs.com), and [Yarn](https://yarnpkg.com/lang/en).
+
+## Initializing
+
+Once you've tried scaffolding out an ordinary React Native project, you'll be ready to start adding TypeScript. Let's go ahead and do that.
+
+```sh
+react-native init MyAwesomeProject
+cd MyAwesomeProject
+```
+
+## Adding TypeScript
+
+The next step is to add TypeScript to your project. The following commands will:
+
+* add TypeScript to your project
+* add [React Native TypeScript Transformer](https://github.com/ds300/react-native-typescript-transformer) to your project
+* initialize an empty TypeScript config file, which we'll configure next
+* add an empty React Native TypeScript Transformer config file, which we'll configure next
+* Adds [typings](https://github.com/DefinitelyTyped/DefinitelyTyped) for React and React Native
+
+Okay let's go ahead and run these.
+
+```sh
+yarn add --dev typescript
+yarn add --dev react-native-typescript-transformer
+yarn tsc --init --pretty --jsx react
+touch rn-cli.config.js
+yarn add --dev @types/react @types/react-native
+```
+
+The `tsconfig.json` file contains all the settings for the TypeScript compile. The defaults created by the command above are mostly fine, but open the file and uncomment the following line:
+
+```js
+{
+  ...
+  // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+  ...
+}
+```
+
+You'll also want to add this to the **top-level** of the config, to tell TypeScript not to try compiling your `node_modules` directory:
+
+```js
+{
+  ...
+  "exclude": [
+    "node_modules"
+  ]
+  ...
+}
+```
+
+The `rn-cli.config.js` contains the settings for the React Native TypeScript Transformer. Open it and add the following:
+
+```js
+module.exports = {
+  getTransformModulePath() {
+    return require.resolve("react-native-typescript-transformer");
+  },
+  getSourceExts() {
+    return ["ts", "tsx"];
+  }
+};
+```
+
+## Migrating to TypeScript
+
+Rename the generated `App.js` and `__tests_/App.js` files to `App.tsx`. `index.js` needs to use the `.js` extension. All new files should use the `.tsx` extension (or `.ts` if the file doesn't contain any JSX).
+
+If you tried to run the app now, you'd get an error like `object prototype may only be an object or null`. This is caused by a failure to import the default export from React as well as a named export on the same line. Open `App.tsx` and modify the import at the top of the file:
+
+```diff
+-import React, { Component } from 'react';
++import React from 'react'
++import { Component } from 'react';
+```
+
+Some of this has to do with differences in how Babel and TypeScript interoperate with CommonJS modules. In the future, the two will stabilize on the same behavior.
```

Comment:
I'm pretty sure you shouldn't need to use this if you use `esModuleInterop`, which I think is appropriate for React Native users.

Reply:
Okay, I'll look into this to verify 👍

Reply:
I took a look at couldn't get this to work. I did find [an open issue](https://github.com/Microsoft/TypeScript/issues/21621) for this, so it should be possible to remove this once TypeScript has been patched to resolve that issue.